### PR TITLE
feat(trs-filer): create TRS-Filer component

### PR DESCRIPTION
**Pull request by ltwmori**
## Description

This PR implements the TRS-filer component as described in the linked issue.

Fixes issue #97 

## Checklist

- [x] My code follows the [contributing guidelines][contributing] of this project.
- [x] I am aware that all my commits will be squashed into a single commit, using the PR title as the commit message.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [x] I have updated the user-facing documentation to describe any new or changed behavior.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have not reduced the existing code coverage.

## Comments

[contributing]: CONTRIBUTING.md

### CODE REVIEW

**Review by SalihuDickson — CHANGES_REQUESTED** [review]
please create a .gitignore file for your package

File: packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/tooltip/tooltip.template.ts
[the remark this answers is not in the document]
Comment (ltwmori) [reply packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/tooltip/tooltip.template.ts:1]: If you had looked more closely at my components, you would have seen that this component is one of the most frequently used by me.

**Comment by SalihuDickson** [comment]
this package doesn't follow the naming format of the other packages, is there a particular reason you named it like this?

**Comment by ltwmori** [comment]
> this package doesn't follow the naming format of the other packages, is there a particular reason you named it like this?

actually I was told to name it like this in the slack. You can refer to the following message https://elixir-cloud.slack.com/archives/C01LU14FZ50/p1684958704099269 --> the second one is `ecc-client-elixir-trs-filer`

File: packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/definition/ecc-trs-filer.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+    },
+    versions: [],
+  };
+
+  /**
+   * Load tool classes from the server
+   * @returns {Promise<void>}
+   * @async
+   */
+  async loadTools() {
```
Comment (SalihuDickson) [inline packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/definition/ecc-trs-filer.ts:45]: you're querying the server for the tools three times whenever the page is refreshed, once here, once in the "trs.list.ts" file and again in the "trs-classes.ts" file, which is inefficient, especially if the data being queried starts to get large. Is there a particular reason you're doing this?

**Review by SalihuDickson — CHANGES_REQUESTED** [review]
great work so far! Let's try to get through these issues so you can merge

**Comment by SalihuDickson** [comment]
in the project folder on your system, go into the "packages/ecc-client-ga4gh-wes" directory and delete the "out-tsc" directory there, it's not supposed to be in the repo, I already deleted it in you branch"

File: packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/copy/copy.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  name: "custom-copy",
+  template,
+  styles,
+})
+export class CopyClipboard extends FASTElement {
+  /** The text value to be copied */
+  @attr value: string;
+
+  /** The text displayed to the user */
+  @observable text: string;
```
[the remark this answers is not in the document]
Comment (ltwmori) [reply packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/copy/copy.ts:24]: resolved

File: packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/copy/copy.template.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+export const template = html<CopyClipboard>`
+  <head>
+    <script
+      type="module"
+      src="https://cdn.jsdelivr.net/npm/@microsoft/fast-components/dist/fast-components.min.js"
+    ></script>
+  </head>
+  <body>
+    <div class="clipboard input">
+      <p>${(x) => x.value}dwdwe</p>
```
[the remark this answers is not in the document]
Comment (ltwmori) [reply packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/copy/copy.template.ts]: resolved

File: packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/tooltip/tooltip.template.ts
[the remark this answers is not in the document]
Comment (ltwmori) [reply packages/ecc-client-elixir-trs-filer/src/components/ecc-trs-filer/custom/tooltip/tooltip.template.ts:1]: resolved

**Review by SalihuDickson — APPROVED** [review]
you can go ahead and merge this and create a new PR for any modifications, this one is already large enough. Great work by the way, while I still have quite concerns we can address in subsequent PRs, your component has come together really well.

**Pull request merged**